pages/index.js

```diff
@@ -1,8 +1,25 @@
 import styled from "styled-components";
 import Head from "next/head";
 import OutivitiesList from "@/components/OutivitiesList";
+import { useState, useEffect } from "react";
+import SearchBar from "@/components/SearchBar";
 
 export default function HomePage({ outivities, favorites, onToggleFavorite }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [foundOutivities, setfoundOutivities] = useState();
+
+  useEffect(() => {
+    const filteredOutivities = outivities.filter((outivity) =>
+      outivity.title.toLowerCase().includes(searchTerm.toLowerCase())
+    );
+
+    const sortedOutivities = filteredOutivities.sort((a, b) =>
+      a.title.localeCompare(b.title)
+    );
+
+    setfoundOutivities(sortedOutivities);
+  }, [searchTerm, outivities]);
+
```

Comment:
I don't think we need `useEffect` here. 

Instead we can use derived state. i.e. Any time search changes, react will re-render the component. Then the value of `foundOutivities` will always be up to date (i.e. derived from the value of `searchTerm`). 

``` js
const foundOutivities = outivities
  .filter((outivity) =>
    outivity.title.toLowerCase().includes(searchTerm.toLowerCase())
  )
  .sort((a, b) => a.title.localeCompare(b.title));

```